torch/csrc/jit/fusion_compiler.cpp

```diff
@@ -590,13 +905,13 @@ struct CUDAFusionFunction : public CompiledFusionFunction {
     checkCUDAVersion(prop);
 
     std::stringstream cu;
-    concat_desc = codegen::emitCompilationUnit(cu, name, agraph, true);
+    concat_desc = codegen::emitCompilationUnit(cu, has_random, name, agraph, true);
     compilation_unit = cu.str();
     nvrtcProgram program;
     TORCH_NVRTC_CHECK(nvrtcCreateProgram(&program, compilation_unit.c_str(), NULL, 0, nullptr, nullptr));
 
     std::string compute = "--gpu-architecture=compute_" + std::to_string(prop.major) + std::to_string(prop.minor);
-    std::vector<const char *> args = {"--std=c++11", compute.c_str()};
+    std::vector<const char *> args = {"--std=c++11", compute.c_str(), "-default-device"};
```

Comment:
Why can't we just annotate our entities with `__device__` as necessary? Is this because of the TF Philox code?

Reply:
Right